### PR TITLE
API wiring to support timestamps.

### PR DESCRIPTION
**Pull request by novemberkilo**

### CODE REVIEW

File: lib/run_time.ml [hunk 1]
```diff
@@ -56,6 +58,19 @@ type run_time_info =
...
+    | Queued v1, Queued v2 -> cmp_floats v1 v2
+    | Running v1, Running v2 -> cmp_floats v1.ready v2.ready && cmp_floats v1.started v2.started
+    | Finished {ready=ready1; started=None; finished=finished1}, Finished {ready=ready2; started=None; finished=finished2} ->
+        cmp_floats ready1 ready2 && cmp_floats finished1 finished2
+    | Finished {ready=ready1; started=Some started1; finished=finished1}, Finished {ready=ready2; started=Some started2; finished=finished2} ->
+        cmp_floats ready1 ready2 && cmp_floats started1 started2 && cmp_floats finished1 finished2
+    | Queued _, (Running _ | Finished _ ) -> false
+    | Running _, (Queued _ | Finished _) -> false
+    | Finished {started=None; _}, (Queued _ | Running _ | Finished {started=Some _; _}) -> false
+    | Finished {started=Some _; _}, (Queued _ | Running _ | Finished {started=None; _}) -> false
```
Comment (maiste) [inline lib/run_time.ml:72]: Couldn't this be written this way, to show that "in other cases it's false"?
```suggestion
    | Queued _, (Running _ | Finished _ )
    | Running _, (Queued _ | Finished _) 
    | Finished {started=None; _}, (Queued _ | Running _ | Finished {started=Some _; _})
    | Finished {started=Some _; _}, (Queued _ | Running _ | Finished {started=None; _}) -> false
```

File: api/client.mli [hunk 2]
```diff
@@ -24,6 +24,9 @@ end
 type job_info = {
   variant : variant;
   outcome : State.t;
+  queued_at : float option;
+  started_at : float option;
+  finished_at : float option;
```
[the remark this answers is not in the document]
Comment (novemberkilo) [reply api/client.mli:29]: This is what I wanted to achieve as well. It is this branch in `Index.get_jobs` that forced this option - essentially it seems that queued jobs may not have a `job_id` in the `ci_build_index` table.

File: api/schema.capnp [hunk 3]
```diff
@@ -34,8 +34,27 @@ struct JobInfo {
     # that the server crashed while building, and when it came back up we
     # no longer wanted to test that commit anyway.
   }
+
+  queuedAt  :union {
+    ts         @6 :Float64;
+    # timestamp as seconds since epoch
+    none       @7 :Void;
```
Comment (tmcgilchrist) [inline api/schema.capnp:41]: These numbers are unique from `@0` based on the `:union` see https://capnproto.org/language.html

File: api/client.mli (hunk 2 above)
[the remark this answers is not in the document]
Comment (novemberkilo) [reply api/client.mli:28]: Also, this is how the pipeline gets the job_id of a job https://github.com/ocurrent/ocaml-ci/blob/master/service/pipeline.ml#L113-L115

[the remark this answers is not in the document]
Comment (maiste) [reply api/client.mli:28]: As it comes from `Index.get_jobs`, it seems fair to keep it this way.

**Review by maiste — APPROVED** [review]
LGTM :+1:

**Comment by novemberkilo** [comment]
Subsumed by #522

**Pull request closed without merging**